# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/forms/nsFieldSetFrame.h
# Commit: 0186165ee1d8
# Full Hash: 0186165ee1d84f56362ba16bbda4eefa1d47fd60
# Author: Mats Palmgren <mats@mozilla.com>
# Date: 2019-11-26 05:19:55
# Regressor Bug: 471015
# File Overlap Count: 1
# Description:
#   Bug 471015 - [css-break] Implement <fieldset> fragmentation.  r=TYLin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D53709
# ==============================================================================

diff -r 9cf1851c76dc -r 0186165ee1d8 layout/forms/nsFieldSetFrame.h
--- a/layout/forms/nsFieldSetFrame.h	Mon Nov 25 15:47:38 2019 +0000
+++ b/layout/forms/nsFieldSetFrame.h	Mon Nov 25 18:05:16 2019 +0000
@@ -16,6 +16,7 @@
 
  public:
   NS_DECL_FRAMEARENA_HELPERS(nsFieldSetFrame)
+  NS_DECL_QUERYFRAME
 
   explicit nsFieldSetFrame(ComputedStyle* aStyle, nsPresContext* aPresContext);
 
@@ -81,13 +82,14 @@
 #endif
 
   /**
-   * Return the anonymous frame that contains all descendants except
-   * the legend frame.  This is currently always a block frame with
-   * pseudo nsCSSAnonBoxes::fieldsetContent() -- this may change in the
-   * future when we add support for CSS overflow for <fieldset>.  This really
-   * can't return null, though callers seem to feel that it can.
+   * Return the anonymous frame that contains all descendants except the legend
+   * frame.  This can be a block/grid/flex/scroll frame.  It always has
+   * the pseudo type nsCSSAnonBoxes::fieldsetContent.  If it's a scroll frame,
+   * the scrolled frame can be a block/grid/flex frame.
+   * This may return null, for example for a fieldset that is a true overflow
+   * container.
    */
-  nsIFrame* GetInner() const;
+  nsContainerFrame* GetInner() const;
 
   /**
    * Return the frame that represents the legend if any.  This may be
@@ -96,8 +98,21 @@
    */
   nsIFrame* GetLegend() const;
 
+  /** @see mLegendSpace below */
+  nscoord LegendSpace() const { return mLegendSpace; }
+
  protected:
+  /**
+   * Convenience method to create a continuation for aChild after we've
+   * reflowed it and got the reflow status aStatus.
+   */
+  void EnsureChildContinuation(nsIFrame* aChild, const nsReflowStatus& aStatus);
+
   mozilla::LogicalRect mLegendRect;
+
+  // This is how much to subtract from our inner frame's content-box block-size
+  // to account for a protruding legend.  It's zero if there's no legend or
+  // the legend fits entirely inside our start border.
   nscoord mLegendSpace;
 };
 